taf/api/utils/_conf.py

```diff
@@ -0,0 +1,39 @@
+from taf.log import taf_logger
+from pathlib import Path
+from typing import Optional
+
+
+def find_taf_directory(auth_repo_path: Path) -> Optional[Path]:
+    """Look for the .taf directory within the archive root.
+
+    Args:
+        auth_repo_path (Path): The path to the authentication repository.
+
+    Returns:
+        Optional[Path]: The path to the .taf directory if found, otherwise None.
+    """
+    # Determine the archive root
+    archive_root = auth_repo_path.parent.parent
+
+    current_dir = archive_root
+    while current_dir != current_dir.root:
+        taf_directory = current_dir / ".taf"
+        if taf_directory.exists() and taf_directory.is_dir():
+            return taf_directory
+        current_dir = current_dir.parent
+
+    taf_logger.info(f"No .taf directory found starting from {archive_root}")
+    return None
+
+
+def find_keystore(path: Path) -> Optional[Path]:
+    """
+    Find keystore starting from the given path and traversing parent directories if needed.
+    """
+    for parent in [path] + list(path.parents):
```

Comment:
Tested and it does not seem like this was able to find the keystore. It's because it's in one of the parent directories / .taf. Tested
```
    path = find_taf_directory(path) / "keystore"
    if path.is_dir():
        return path
``` 
And that worked